ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -289,22 +298,44 @@ func (ipfs *Connector) Pin(ctx context.Context, hash cid.Cid, maxDepth int) erro
 		pinArgs = fmt.Sprintf("recursive=true&max-depth=%d", maxDepth)
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	timer := time.NewTimer(ipfs.config.PinTimeout)
+	defer timer.Stop()
+
+	reset := make(chan int)
```

Comment:
channel never closed. Also, `make(chan struct{})`

Reply:
> Also, make(chan struct{})

why?

Reply:
https://dave.cheney.net/2014/03/25/the-empty-struct